website/static/js/share/results.js

```diff
@@ -37,79 +38,155 @@ Results.controller = function(vm) {
     self.vm = vm;
     self.vm.resultsLoading = m.prop(false);
 
+    self.renderTitleBar = function(result) {
```

Comment:
Instead of making these controller methods, why not implement these as proper mithril components? No need to reinvent the wheel for composing UI elements.

```js
var Titlebar = {
    view: function(ctrl, params) {
        var result = params.result;
        return m('span', {}, [
            m('a[href=' + result.uris.canonicalUri + ']', ((result.title || '').length > 0) ? m.trust(result.title) : 'No title provided'),
            m('br'),
            m.component(Description, {result: result})
        ]);
    }
};

// ...

var Results = {
    view: function(ctrl, params) {
        var resultViews = $.map(ctrl.vm.results, function(result, i) {
            return m.component(Result, {result: result});
        });
        returm m(..., ..., [resultViews, ...])
    }
}
```